pgmuvi/lightcurve.py

```diff
@@ -355,13 +365,38 @@ def yerr(self):
 
     @yerr.setter
     def yerr(self, values):
+        # first, check that the input is a tensor
+        # and modifiy it if necessary
+        values = self._ensure_tensor(values)
+        # then, store the raw data internally
         self._yerr_raw = values
         # now apply the same transformation that was applied to the ydata
         if self.ytransform is None:
             self._yerr_transformed = values
         elif isinstance(self.ytransform, Transformer):
             self._yerr_transformed = self.ytransform.transform(values)
 
+    def _ensure_tensor(self, values):
+        # Ensures that the input data has type torch.Tensor
+        # Transforms the data if necessary
+        if not isinstance(values, torch.Tensor):
+            warnings.warn(('The function expects a torch.Tensor as input.'
+                           'Your data will be converted to a tensor.'),
+                          stacklevel=2)
+            values = torch.as_tensor(values, dtype=torch.float32)
```

Comment:
Could we change the dtype here to be more general? I can see float32 being the default, but maybe we should have a self._default_dtype to cast to, which defaults to torch.float32? Again, this isn't a showstopper.

Reply:
Sure, I'll think about it.